File: app/src/main/java/com/sudhindra/spiderinductionstask2/apis/APODApi.java
package com.sudhindra.spiderinductionstask2.apis;

import com.sudhindra.spiderinductionstask2.models.APOD;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface APODApi {

    @GET("apod")
    Call<APOD> getAPODForToday(@Query("api_key") String apiKey);

    @GET("apod")
    Call<APOD> getAPOD(@Query("date") String date, @Query("api_key") String apiKey);
}

File: app/src/main/java/com/sudhindra/spiderinductionstask2/apis/APODApi.kt
package com.sudhindra.spiderinductionstask2.apis

import com.sudhindra.spiderinductionstask2.models.APOD
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface APODApi {
    @GET("apod")
    fun getAPODForToday(@Query("api_key") apiKey: String?): Call<APOD?>?

    @GET("apod")
    fun getAPOD(@Query("date") date: String?, @Query("api_key") apiKey: String?): Call<APOD?>?
}
